packages/core/src/machines/actors/auth/signIn.ts

```diff
@@ -0,0 +1,280 @@
+import { createMachine, sendParent, assign, sendUpdate } from 'xstate';
+import { get } from 'lodash';
+
+import { AuthEvent, AuthChallengeNames, SignInContext } from '../../../types';
+import { Auth } from 'aws-amplify';
+
+export const signInActor = createMachine<SignInContext, AuthEvent>(
+  {
+    initial: 'init',
+    id: 'signInActor',
+    context: {
+      remoteError: '',
+      formValues: {},
+      validationError: {},
+    },
+    states: {
+      init: {
+        always: [
+          { target: 'autoSignIn', cond: 'shouldAutoSignIn' },
+          { target: 'signIn' },
+        ],
+      },
+      autoSignIn: {
+        invoke: {
+          src: 'signIn',
+          onDone: { actions: 'setUser', target: '#signInActor.resolved' },
+          onError: '#signInActor.signIn',
+        },
+      },
+      signIn: {
+        initial: 'edit',
+        states: {
+          edit: {
+            entry: sendUpdate(),
+            on: {
+              SUBMIT: 'submit',
+              CHANGE: { actions: 'handleInput' },
+              FEDERATED_SIGN_IN: 'federatedSignIn',
+            },
+          },
+          federatedSignIn: {
```

Comment:
Now, federatedSignIn is under signIn flow as its goal matches that of signIn actor. This means that there will be some dry code in signUp.federatedSignUp, but we can later reuse code with shared state definition + spread operator.